### PR TITLE
Service Fabric knack conversion

### PR DESCRIPTION
**Pull request by derekbekoe**
Implements #4940.

### CODE REVIEW

**Comment by tjprescott** [comment]
@derekbekoe please annotate #4940 with the fact that the automated tests are being skipped.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/__init__.py [hunk 1]
```diff
@@ -3,12 +3,28 @@
...
 # --------------------------------------------------------------------------------------------
 
+from azure.cli.core import AzCommandsLoader
+
 import azure.cli.command_modules.servicefabric._help  # pylint: disable=unused-import
 
 
-def load_params(_):
-    import azure.cli.command_modules.servicefabric._params   # pylint: disable=redefined-outer-name, unused-variable
+class SFCommandsLoader(AzCommandsLoader):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/__init__.py]: I'd recommend ServiceFabricCommandsLoader

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py [hunk 2]
```diff
@@ -3,127 +3,101 @@
...
+def load_arguments(self, _):  # pylint: disable=too-many-statements
+    with self.argument_context('sf') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type, id_part=None, help='The resource group name')
+        c.argument('cluster_name', options_list=['--name', '--cluster-name', '-n'], help='Specify the name of the cluster, if not given it will be same as resource group name')
+        c.argument('location', validator=get_default_location_from_resource_group)
+        c.argument('secret_identifier', options_list=['--secret-identifier'], help='The existing Azure key vault secret URL')
+        c.argument('certificate_file', options_list=['--certificate-file'], help='The existing certificate file path for the primary cluster certificate.')
+        c.argument('parameter_file', options_list=['--parameter-file'], help='The path to the template parameter file.')
+        c.argument('template_file', options_list=['--template-file'], help='The path to the template file.')
+        c.argument('vm_password', options_list=['--vm-password'], help='The password of the Vm')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py]: Most of these options_lists can be removed as they are exactly what you'd get from the dest name conversion.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py [hunk 3]
```diff
@@ -3,127 +3,101 @@
...
+        c.argument('number_of_nodes_to_remove', help='number of nodes to remove.')
+
+    with self.argument_context('sf cluster durability') as c:
+        c.argument('durability_level', arg_type=get_enum_type(['Bronze', 'Silver', 'Gold']), help='durability level.')
+
+    with self.argument_context('sf cluster setting') as c:
+        c.argument('parameter', help='parameter name')
+        c.argument('section', help='section name')
+        c.argument('value', help='Specify the value')
+        c.argument('settings_section_description', help='Specify the value')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py]: "Specify the value"?  Should be something better here.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/__init__.py (hunk 1 above)
Comment (derekbekoe) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/__init__.py]: ok done.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py (hunk 2 above)
Comment (derekbekoe) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py]: true. fixed.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py (hunk 3 above)
Comment (derekbekoe) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py]: Leaving as is as that's what they have currently.

**Pull request merged**